pg_lakehouse/README.md

```diff
@@ -0,0 +1,299 @@
+
+`pg_lakehouse` uses the foreign data wrapper (FDW) API to connect to any object store or table format and the executor hook API to push queries to DataFusion. While other FDWs like `aws_s3` have existed in the Postgres extension ecosystem, these FDWs have two major limitations:
+
+1. Lack of support for most object stores, file, and table formats
+2. Too slow over large datasets to be a viable analytical engine
+
+`pg_lakehouse` differentiates itself by supporting a wide breadth of stores and formats (thanks to [OpenDAL](https://github.com/apache/opendal)) and by being very fast (thanks to [DataFusion](https://github.com/apache/datafusion)).
+
+## Getting Started
+
+The following example uses `pg_lakehouse` to query an example dataset of 3 million NYC taxi trips from January 2024, hosted in a public S3 bucket provided by ParadeDB.
+
+```sql
+CREATE EXTENSION pg_lakehouse;
+CREATE FOREIGN DATA WRAPPER s3_wrapper HANDLER s3_fdw_handler VALIDATOR s3_fdw_validator;
+
+-- Provide S3 credentials
+CREATE SERVER s3_server FOREIGN DATA WRAPPER s3_wrapper
+OPTIONS (bucket 'paradedb-benchmarks', region 'us-east-1', allow_anonymous 'true');
+
+-- Create foreign table
+CREATE FOREIGN TABLE trips (
+    "VendorID"              INT,
+    "tpep_pickup_datetime"  TIMESTAMP,
+    "tpep_dropoff_datetime" TIMESTAMP,
+    "passenger_count"       BIGINT,
+    "trip_distance"         DOUBLE PRECISION,
+    "RatecodeID"            DOUBLE PRECISION,
+    "store_and_fwd_flag"    TEXT,
+    "PULocationID"          REAL,
+    "DOLocationID"          REAL,
+    "payment_type"          DOUBLE PRECISION,
+    "fare_amount"           DOUBLE PRECISION,
+    "extra"                 DOUBLE PRECISION,
+    "mta_tax"               DOUBLE PRECISION,
+    "tip_amount"            DOUBLE PRECISION,
+    "tolls_amount"          DOUBLE PRECISION,
+    "improvement_surcharge" DOUBLE PRECISION,
+    "total_amount"          DOUBLE PRECISION
+)
+SERVER s3_server
+OPTIONS (path 's3://paradedb-benchmarks/yellow_tripdata_2024-01.parquet', extension 'parquet');
+
+-- Success! Now you can query the remote Parquet file like a regular Postgres table
+SELECT COUNT(*) FROM trips;
+  count
+---------
+ 2964624
+(1 row)
+```
+
+Note that column names must be wrapped in double quotes to preserve uppercase letters. This is because DataFusion is case-sensitive and Postgres' foreign table column names must match the foreign table's column names exactly.
+
+## Query Acceleration
+
+This extension uses Postgres hooks to intercept and push queries down to DataFusion. In order to enable these hooks, the extension
+must be added to `shared_preload_libraries` inside `postgresql.conf`. If you are using Postgres 16, this file can be found under `~/.pgrx/data-16`.
```

Comment:
"using Postgres 16 and `pgrx`"? What if they're not using pgrx

Reply:
Sure, this was copy pasted from pg_search